projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -315,6 +307,290 @@ private boolean answer(String questionTemplateName, String paramsLine,
+            if (!value.isTextual()) {
+               throw new BatfishException(
+                     String.format("A Batfish %s must be a JSON string",
+                           expectedType.getName()));
+            }
+            // TODO: need prefix range validation check
+            break;
+         case JSON_PATH:
+            validateJsonPath(value);
+            break;
+         default:
+            throw new BatfishException(
+                  String.format("Unsupported parameter type: %s",
+                        expectedType));
+      }
+   }
+
+   /**
+    * Validate that {@code jsonPathRegex} contains a valid Java regular
+    * expression of a {@code JsonPath} (Starts with "/", ends with either "/" or
+    * "/i", contains a valid Java regular expression between "/").
+    *
+    * <p> As written, this function will accept the strings "/" and "/i" as
+    * complete expressions – resulting in an empty inner Java regular
+    * expression.</p>
+    *
+    * @throws BatfishException if the content of {@code jsonPathRegex} is not
+    * a valid Java regular expression of a JsonPath.
+    */
+   static void validateJsonPathRegex(String jsonPathRegex)
+         throws BatfishException {
+      if (!jsonPathRegex.startsWith("/")) {
+         throw new BatfishException(
+               String.format("A Batfish %s must start with %s",
+                              Variable.Type.JSON_PATH_REGEX.getName(),
+                              "\"/\""));
+      }
+      if (!(jsonPathRegex.endsWith("/") || jsonPathRegex.endsWith("/i"))) {
+         throw new BatfishException(
+               String.format("A Batfish %s must end in either %s or %s",
+                     Variable.Type.JSON_PATH_REGEX.getName(),
+                     "\"/\"", "\"/i\""));
+      }
+      String innerPath = "";
+      if (jsonPathRegex.lastIndexOf('/') > 0) {
+         innerPath =
+               jsonPathRegex.substring(1, jsonPathRegex.lastIndexOf('/'));
+      }
+      try {
+         Pattern.compile(innerPath);
+      } catch (PatternSyntaxException e) {
+         throw new BatfishException(
+               String.format("Invalid %s at interior of %s",
+                     Variable.Type.JAVA_REGEX.getName(),
+                     Variable.Type.JSON_PATH_REGEX.getName()), e);
+      }
+   }
+
+   /**
+    * Validate that json-encoded {@code jsonPath} is a valid jsonPath dictionary
+    * (A valid jsonPath contains key 'path' which mapping to a String, and an
+    * optional key 'suffix' which mapping to a boolean value).
+    *
+    * @throws BatfishException if {@code jsonPath} is not a valid jsonPath
+    * dictionary.
+    */
+   static void validateJsonPath(JsonNode jsonPath) throws BatfishException {
+      if (!jsonPath.isContainerNode()) {
+         throw new BatfishException(
+               String.format("%s is not a valid type for a Batfish %s. Type " +
+                           "expected: JSON dictionary", jsonPath.getNodeType(),
```

Comment:
Is the return value of  `jsonPath.getNodeType()` something that a user that does not use Jackson know about?